lib/Project.js

```diff
@@ -232,6 +232,13 @@ Project.prototype._discoverTypes = function (proto, package) {
     var eName = helper.joinPackage(package, e.getName())
     this._typesByName[eName] = e
   }, this)
+
+  if (!proto.getServices) {
```

Comment:
`proto` might be a `MessageDescriptor`, and `.getServices` is only present when `proto` is a `ProtoDescriptor`
```
@param {MessageDescriptor|ProtoDescriptor} proto
```

the function recurses on `message` (see `_discoveryTypes(..)` at L226) to resolve nested message types.  but we don't support nesting `service` definitions within `messages`, so there's no `.getServices` on the `MessageDescriptor`.